<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="topic_from_controller" value="/follower/control_effort"/>
    <arg name="topic_from_plant" value="/follower/state"/>
    <arg name="setpoint_topic" value="/follower/setpoint"/>
    <node name="follower" pkg="rr_common" type="follower" output="screen">
        <param name="MIN_FRONT_VISION" value="0.37"/>
        <param name="MAX_FRONT_VISION" value="3"/>
        <param name="GOAL_DIST" value="1.3"/>
        <param name="GOAL_MARGIN_OF_ERR" value="0.2"/>
        <param name="MIN_SIDE_VISION" value="-0.5"/>
        <param name="MAX_SIDE_VISION" value="0.5"/>
        <param name="FOLLOWER_SPEED" value="0.2"/>
        <param name="INPUT_CLOUD_TOPIC" value="/scan/pointcloud"/>
        <param name="topic_from_controller" value="$(arg topic_from_controller)" />
        <param name="topic_from_plant" value="$(arg topic_from_plant)" />
        <param name="setpoint_topic" value="$(arg setpoint_topic)" />
    </node>
    <node name="demo_pid_controller" pkg="pid" type="controller" >
        <param name="Kp" value="-2.0" />
        <param name="Ki" value="-0.8" />
        <param name="Kd" value="-0.16" />
        <param name="upper_limit" value="1" />
        <param name="lower_limit" value="-1" />
        <param name="windup_limit" value="10" />
        <param name="topic_from_controller" value="$(arg topic_from_controller)" />
        <param name="topic_from_plant" value="$(arg topic_from_plant)" />
        <param name="setpoint_topic" value="$(arg setpoint_topic)" />
    </node>
</launch>
